### ISSUE TITLE
Test improvement

### ISSUE THREAD

**Issue by henrywang** [containers/bootc#868]
I went through all tests in this repo. We have unit test, integration test and end to end test. The tests running against each PR should be fast and stable. And test should be easy to make contribution. I have some thoughts on test:

- Keep running unit test, it's fast and easy to make contribution. Enable code coverage?
- end to end test can be removed because it takes long time to run and QE has a daily end to end test against latest `main` copr build.
- Integration test should have all bootc commands and args included.
- Now we have `nushell` and `python` for integration test. I like `nushell` a lot. But looks `python` is generic for test code.
- TMT + Packit is good for Fedora and CentOS, but it does not support the other distros.

The reason I am opening this issue is to start a discussion about testing. Comments are welcome!

**Comment by cgwalters** [comment]
> end to end test can be removed because it takes long time to run and QE has a daily end to end test against latest main copr build.

Yes, though I'd like to be able to opt-in to it for specific PRs; I think we can do that with a label trigger or a comment trigger.

> Now we have nushell and python for integration test. I like nushell a lot. But looks python is generic for test code.

Yeah let's kill the python stuff.

> TMT + Packit is good for Fedora and CentOS, but it does not support the other distros.

Yes, though for the most part just *building* across other distros/OSes is trivial in containers in GHA; the harder part is reference base images and builds.

**Comment by cgwalters** [comment]
This one was a bit vague but I think it's done. We should have dedicated followup issues for specific topics.
